sql/src/main/java/io/crate/rest/action/RestSQLAction.java

```diff
@@ -149,11 +153,43 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
 
     @VisibleForTesting
     User userFromRequest(RestRequest request) {
-        String user = request.header(AuthSettings.HTTP_HEADER_USER);
-        if (user == null) {
-            user = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.setting().get(settings);
+        String username = extractUsernameFromHttpBasicAuthHeader(request.header(AuthSettings.HTTP_HEADER_BASIC_AUTH));
+
+        // Fallback to deprecated setting
+        if (username == null) {
+            username = request.header(AuthSettings.HTTP_HEADER_USER);
+        }
+
+        // Fallback to trusted user from configuration
+        if (username == null) {
+            username = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.setting().get(settings);
+        }
+        return userManager.findUser(username);
+    }
+
+    private static String extractUsernameFromHttpBasicAuthHeader(String authHeader) {
```

Comment:
authHeader is a very confusing name
initially I thought this would be the header name
you can leave the name if you document that this is the **content** of the header